src/main/webapp/script.js

```diff
@@ -39,7 +39,30 @@ function displayError() {
 
 function getZipCode() {
   let zip = document.getElementById('zipCode').value;
-  fetchList('/data?zipCode=' + zip);
+  if (isValidInput(zip)) {
+    fetchList('/data?zipCode=' + zip);
+  }
+  else {
+    window.alert('Invalid input. Try again.'); 
+  }
+}
+
+function isValidInput(zip) {
+  let len = zip.length;
+  if (len === 0) {  
+    return false; 
+  }
+  for (let i=0; i<len; i++) {
```

Comment:
nit: Please add a space above this line.

Reply:
Fixed!